packages/server/src/errors.ts

```diff
@@ -299,39 +159,60 @@ export function formatApolloErrors(
   //   flattenedErrors.push(error);
   // }
 
-  const enrichedErrors = errors.map((error) => enrichError(error, debug));
-  const makePrintable = (error: GraphQLFormattedError) => {
-    if (error instanceof Error) {
-      // Error defines its `message` and other fields as non-enumerable, meaning JSON.stringify does not print them.
-      const graphQLError = error as GraphQLFormattedError;
-      return {
-        message: graphQLError.message,
-        ...(graphQLError.locations && { locations: graphQLError.locations }),
-        ...(graphQLError.path && { path: graphQLError.path }),
-        ...(graphQLError.extensions && { extensions: graphQLError.extensions }),
-      };
-    }
-    return error;
-  };
-
-  if (!formatter) {
-    return enrichedErrors;
-  }
+  const { includeStackTracesInErrorResponses, errorCode } = options;
 
-  return enrichedErrors.map((error) => {
+  const formatError = options.formatError ?? ((error) => error);
+  return errors.map((error) => {
     try {
-      return makePrintable(formatter(error));
+      return formatError(enrichError(error), error);
```

Comment:
Interesting note: the change to two-arg formatError actually makes the fact that formatError is integrated into this function rather than just a second `map` over its output more crucial.
(ie, I was going to propose that (perhaps in a follow-up PR) we just remove this option and have the callers call formatError instead (or totally drop formatError and just have callers use didEncounterErrors from the plugin API) but this new argument makes the current structure make more sense.)